File: src/tools/clear.ts
import { z } from "zod";
import { TerminalManager } from "../terminal/index.js";

export const clearSchema = z.object({});

export type ClearArgs = z.infer<typeof clearSchema>;

export const clearTool = {
  name: "clear",
  description: "Clear the terminal screen and scrollback buffer.",
  inputSchema: {
    type: "object" as const,
    properties: {},
    required: [],
  },
};

export function handleClear(manager: TerminalManager, _args: unknown): { content: Array<{ type: "text"; text: string }> } {
  manager.clear();

  return {
    content: [
      {
        type: "text",
        text: "Terminal cleared",
      },
    ],
  };
}

File: src/tools/clear.rs
use serde::Deserialize;
use serde_json::Value;

use crate::mcp::{CallToolResult, ToolDefinition};
use crate::terminal::Terminal;

/// Arguments for the clear tool (currently none)
#[derive(Debug, Deserialize, Default)]
pub struct ClearArgs {}

/// Clear the terminal screen
pub fn clear(terminal: &mut Terminal, args: Value) -> Result<CallToolResult, String> {
    let _args: ClearArgs = serde_json::from_value(args).unwrap_or_default();

    terminal
        .clear()
        .map_err(|e| format!("Failed to clear terminal: {}", e))?;

    Ok(CallToolResult::text("Terminal cleared"))
}

/// Get the tool definition for clear
pub fn definition() -> ToolDefinition {
    ToolDefinition {
        name: "clear".to_string(),
        description: "Clear the terminal screen and move cursor to the top-left position.".to_string(),
        input_schema: serde_json::json!({
            "type": "object",
            "properties": {}
        }),
    }
}
